File: Tarea8/src/Libreria.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Random;
public class Libreria {
    private ArrayList<Usuario> usuarios = new ArrayList<>();
    private ArrayList<Libro> libros = new ArrayList<>();

    Scanner sc = new Scanner(System.in);
    Random random = new Random();

    public String mostrarEmpleados() {
        if (usuarios.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("Agregue un usuario con la opción '3'");
        } else {
            int i = 0;
            for (Usuario usuario : usuarios) {
                System.out.printf("Usuario %d:\n", i += 1);
                System.out.printf("%s\n",  usuario.getDatos());
            }
        }
        return ("");
    }
    public String mostrarLibros() {
        if (libros.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay libros registrados");
        } else {
            int i = 0;
            for (Libro libro : libros) {
                System.out.printf("Libro %d:\n", i += 1);
                System.out.printf("%s\n", libro.getDatos());
            }
        }
        return ("");
    }
    public void agregarUsuarios() {
        int id = random.nextInt(100000, 999999);
        System.out.println("-Nombre:");
        String name = sc.next();
        System.out.println("-Edad:");
        int edad = sc.nextInt();
        Usuario usuario = new Usuario(name, id, edad);
        usuarios.add(usuario);
        System.out.println("Se han registrado con éxito los datos");
        System.out.println("****************");
    }
    public void agregarLibros() {
        int id = random.nextInt(100000, 999999);
        System.out.println("-Nombre:");
        String name = sc.next();
        System.out.println("-Autor:");
        String autor = sc.next();
        Libro libro = new Libro(name, autor, id);
        libros.add(libro);
        System.out.println("Se han registrado con éxito los datos");
        System.out.println("****************");
    }
    public String rentarLibros(){
        if(libros.isEmpty()){
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay libros para rentar.");
        }else{
            System.out.println("¿Que empleado es el que va a rentar un libro");
        mostrarEmpleados();
        int i = sc.nextInt();
        if (i > usuarios.size()) {
            System.out.println("No se ha encontrado el empleado.");
        } else {
            System.out.println("¿Que libro le gustaria rentar?:");
            mostrarLibros();
            int j = sc.nextInt();
            if (j > libros.size()) {
                System.out.println("No se ha encontrado el libro ingresado.");
            } else {
                Libro libro = libros.get(j - 1);
                if (libro.isRentado()==true) {
                    System.out.println("El libro ya ha sido rentado.");
                } else {
                    libro.setRentado(true);
                    Usuario usuario = usuarios.get(i - 1);
                    usuario.addLibro(libro);
                    System.out.println("Libro rentado con éxito.");
            }
        }}}
        return ("");
    }
    public String mostrarLibrosRentados() {
        if (libros.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay libros registrados");
        } else {
            int i = 0;
            System.out.println("Libros rentados:");
            for (Libro libro : libros) {
                if (libro.isRentado()==true) {
                System.out.printf("Libro %d:\n", i += 1);
                System.out.printf("%s\n", libro.getDatos());}
            }
        }
        return ("");
    }
    public String mostrarLibrosDisponibles() {
        if (libros.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay libros registrados");
        } else {
            int i = 0;
            System.out.println("Libros disponibles:");
            for (Libro libro : libros) {
                if (libro.isRentado()==false) {
                    System.out.printf("Libro %d:\n", i += 1);
                    System.out.printf("%s\n", libro.getDatos());}
            }
        }
        return ("");
    }
    public String mostrarUsuariosLibros(){
        if (usuarios.isEmpty()){
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("Ningun usuario ha rentado un libro");
        }else{
            int i = 0;
           for (Usuario usuario : usuarios){
               System.out.println("Libros rentados por el usuario");
               System.out.printf("Usuario %d:\n", i += 1);
               System.out.printf("%s\n",  usuario.getDatos());
               System.out.println(usuario.getLibrosRentados());
               System.out.println("***********************");
           }
        }
        return ("");
    }
}

File: Tarea7_Python/Libreria.py
from random import randint
from Usuario import Usuario
from Libro import Libro


class Libreria:
  __usuarios=[]
  __libros=[]

  def __init__(self):
    self.__usuarios = []
    self.__libros = []

  def mostrar_empleados(self):
    if not self.__usuarios:
        print("*****  V  A  C  I  O  ******")
        print("Agregue un usuario con la opción '3'")
    else:
        i = 0
        for usuario in self.__usuarios:
            print(f"Usuario {i + 1}:")
            print(usuario.get_datos())
            i += 1

  def mostrar_libros(self):
    if not self.__libros:
        print("*****  V  A  C  I  O  ******")
        print("No hay libros registrados")
    else:
        i = 0
        for libro in self.__libros:
            print(f"Libro {i + 1}:")
            print(libro.get_datos())
            i += 1

  def agregar_usuarios(self):
    id = randint(100000, 999999)
    print("-Nombre:")
    name = input()
    print("-Edad:")
    edad = int(input())
    usuario = Usuario(name, id, edad)
    self.__usuarios.append(usuario)
    print("Se han registrado con éxito los datos")
    print("****************")

  def agregar_libros(self):
    id =randint(100000, 999999)
    print("-Nombre:")
    name = input()
    print("-Autor:")
    autor = input()
    libro = Libro(name, autor, id)
    self.__libros.append(libro)
    print("Se han registrado con éxito los datos")
    print("****************")

  def rentar_libros(self):
    if not self.__libros:
        print("*****  V  A  C  I  O  ******")
        print("No hay libros para rentar.")
    else:
        print("¿Qué empleado es el que va a rentar un libro?")
        self.mostrar_empleados()
        i = int(input())
        if i > len(self.__usuarios):
            print("No se ha encontrado el empleado.")
        else:
            print("¿Qué libro le gustaría rentar?")
            self.mostrar_libros()
            j = int(input())
            if j > len(self.__libros):
                print("No se ha encontrado el libro ingresado.")
            else:
                libro = self.__libros[j - 1]
                if libro.es_rentado():
                    print("El libro ya ha sido rentado.")
                else:
                    libro.set_rentado(True)
                    usuario = self.__usuarios[i - 1]
                    usuario.add_libro(libro)
                    print("Libro rentado con éxito.")

  def mostrar_libros_rentados(self):
    if not self.__libros:
        print("*****  V  A  C  I  O  ******")
        print("No hay libros registrados")
    else:
        i = 0
        print("Libros rentados:")
        for libro in self.__libros:
            if libro.es_rentado():
                print(f"Libro {i + 1}:")
                print(libro.get_datos())
            i += 1

  def mostrar_libros_disponibles(self):
    if not self.__libros:
        print("*****  V  A  C  I  O  ******")
        print("No hay libros registrados")
    else:
        i = 0
        print("Libros disponibles:")
        for libro in self.__libros:
            if not libro.es_rentado():
                print(f"Libro {i + 1}:")
                print(libro.get_datos())
            i += 1

  def mostrar_usuarios_libros(self):
    if not self.__usuarios:
        print("*****  V  A  C  I  O  ******")
        print("Ningún usuario ha rentado un libro")
    else:
        i = 0
        for usuario in self.__usuarios:
            print("Libros rentados por el usuario")
            print(f"Usuario {i + 1}:")
            print(usuario.get_datos())
            print(usuario.get_libros_rentados())
            print("***********************")
            i += 1